src/plugins/ui_core_views/row_size.ts

```diff
@@ -0,0 +1,164 @@
+        let sizes = [...this.sizes[cmd.sheetId]];
+        sizes = removeIndexesFromArray(sizes, cmd.elements);
+        this.history.update("sizes", cmd.sheetId, sizes);
+        break;
+      }
+      case "ADD_COLUMNS_ROWS": {
+        if (cmd.dimension === "COL") {
+          return;
+        }
+        const oldSizes = [...this.sizes[cmd.sheetId]];
+        const addIndex = getAddHeaderStartIndex(cmd.position, cmd.base);
+        const newRows = Array(cmd.quantity).fill(DEFAULT_CELL_HEIGHT);
+        const newSizes = insertItemsAtIndex(oldSizes, newRows, addIndex);
+        this.history.update("sizes", cmd.sheetId, newSizes);
+        break;
+      }
+      case "RESIZE_COLUMNS_ROWS":
+        {
+          const sheetId = cmd.sheetId;
+          if (cmd.dimension === "ROW") {
+            for (let el of cmd.elements) {
+              this.history.update("sizes", sheetId, el, this.getRowComputedSize(sheetId, el));
+            }
+          } else {
+            // Recompute row heights on col size change, they might have changed because of wrapped text
+            for (let row of range(0, this.getters.getNumberRows(sheetId))) {
+              this.history.update("sizes", sheetId, row, this.getRowComputedSize(sheetId, row));
+            }
+          }
+        }
+        break;
+      case "UPDATE_CELL":
+        const { sheetId, row } = cmd;
+        if (!this.getters.getUserRowSize(sheetId, row)) {
+          if (!this.sizes[sheetId]) debugger;
+          this.history.update("sizes", sheetId, row, this.getRowComputedSize(sheetId, row));
+        }
+        break;
+      case "ADD_MERGE":
+      case "REMOVE_MERGE":
+        const changerRows = new Set<number>();
+        for (let target of cmd.target) {
+          for (let row of range(target.top, target.bottom + 1)) {
+            changerRows.add(row);
+          }
+        }
+
+        for (let row of changerRows) {
+          this.history.update("sizes", cmd.sheetId, row, this.getRowComputedSize(cmd.sheetId, row));
+        }
+        break;
+    }
+    return;
+  }
+
+  getRowSize(sheetId: UID, row: HeaderIndex): Pixel {
+    if (!this.sizes[sheetId]) debugger;
+    return Math.round(
+      this.getters.getUserRowSize(sheetId, row) ?? this.sizes[sheetId][row] ?? DEFAULT_CELL_HEIGHT
+    );
+  }
+
+  private computeRowSizes(sheetId: UID): Array<Pixel> {
+    const sizes: Pixel[] = [];
+    for (let row = 0; row < this.getters.getNumberRows(sheetId); row++) {
+      sizes.push(this.getRowComputedSize(sheetId, row));
+    }
+    return sizes;
+  }
+
+  /**
+   * Return the height the cell should have in the sheet, which is either DEFAULT_CELL_HEIGHT if the cell is in a multi-row
+   * merge, or the height of the cell computed based on its font size.
+   */
+  private getCellHeight(position: CellPosition): Pixel {
+    const merge = this.getters.getMerge(position);
+    if (merge && merge.bottom !== merge.top) {
+      return DEFAULT_CELL_HEIGHT;
+    }
```

Comment:
By reading this, I don't understand why it's should be the default value if the cell is inside a merge.

EDIT: Ah I better understand now that I read the docstring of the other function below. Maybe it managing merges inside `getRowComputedSize` would make it easier to understand